File: Server/PRMServer.Application/DTOs/Decks/DeckSummaryDTO.cs
using PRMServer.Application.DTOs.Users;

namespace PRMServer.Application.DTOs.Decks
{
    public record DeckSummaryDTO
    {
        public required int Id { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public required bool IsPublic { get; set; }
        public required int Version { get; set; }
        public required int ViewsTotal { get; set; }
        public required int ViewsWeekly { get; set; }
        public required int DownloadsTotal { get; set; }
        public required int DownloadsWeekly { get; set; }
        public required DateTime CreatedAt { get; set; }
        public required int CardsCount { get; set; }
        public required UserSummaryDTO Creator { get; set; }
    }
}

File: App/app/src/main/java/com/prm392g2/prmapp/api/dtos/decks/DeckSummaryDTO.java
package com.prm392g2.prmapp.api.dtos.decks;

import com.prm392g2.prmapp.api.dtos.users.UserSummaryDTO;

import java.util.GregorianCalendar;

public class DeckSummaryDTO
{
    public int id;
    public String name;
    public String description;
    public boolean isPublic;
    public int version;
    public int viewsTotal;
    public int viewsWeekly;
    public int downloadsTotal;
    public int downloadsWeekly;
    public GregorianCalendar createdAt;
    public int cardsCount;
    public UserSummaryDTO creator;

}
